src/credentials/CertificationDeclaration.h

```diff
@@ -62,6 +63,36 @@ struct CertificationElements
     bool DACOriginVIDandPIDPresent;
 };
 
+class CertificationElementsDecoder
+{
+public:
+    uint16_t FormatVersion;
+    uint16_t VendorId;
+    bool ContainsPID;
+    uint32_t DeviceTypeId;
+    char CertificateId[kCertificateIdLength + 1];
+    uint8_t SecurityLevel;
+    uint16_t SecurityInformation;
+    uint16_t VersionNumber;
+    uint8_t CertificationType;
+    uint16_t DACOriginVendorId;
+    uint16_t DACOriginProductId;
+    bool DACOriginVIDandPIDPresent;
```

Comment:
Please move these to a struct called `CertificationDeclarationContent` similar to what you did for `DeviceInfoForAttestation`. Please drop the starting capital letters.

You can then have a member struct, and return a const reference to it after properly reading.

Reply:
Fixed in 9cddb31